File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/Tva.Core/Tva.Net/Ftp/SessionDisconnected.vb
' James Ritchie Carroll - 2003
Option Explicit On 

Imports System.Text

Namespace Net.Ftp

    Friend Class SessionDisconnected

        Implements ISessionState

        Private m_host As Session
        Private m_server As String
        Private m_port As Integer
        Private m_caseInsensitive As Boolean

        Friend Sub New(ByVal h As Session, ByVal CaseInsensitive As Boolean)

            m_port = 21
            m_host = h
            m_caseInsensitive = CaseInsensitive

        End Sub

        Public Property Server() As String Implements ISessionState.Server
            Get
                Return m_server
            End Get
            Set(ByVal Value As String)
                m_server = Value
            End Set
        End Property

        Public Property Port() As Integer Implements ISessionState.Port
            Get
                Return m_port
            End Get
            Set(ByVal Value As Integer)
                m_port = Value
            End Set
        End Property

        Public Sub Connect(ByVal UserName As String, ByVal Password As String) Implements ISessionState.Connect

            Dim ctrl As New ControlChannel(m_host)

            ctrl.Server = m_server
            ctrl.Port = m_port
            ctrl.Connect()

            Try
                ctrl.Command("USER " & UserName)

                If ctrl.LastResponse.Code = Response.UserAcceptedWaitingPass Then
                    ctrl.Command("PASS " & Password)
                End If

                If ctrl.LastResponse.Code <> Response.UserLoggedIn Then
                    Throw New AuthenticationException("Failed to login.", ctrl.LastResponse)
                End If

                m_host.State = New SessionConnected(m_host, ctrl, m_caseInsensitive)
                CType(m_host.State, SessionConnected).InitRootDirectory()
            Catch
                ctrl.Close()
                Throw
            End Try

        End Sub

        Public Property CurrentDirectory() As Directory Implements ISessionState.CurrentDirectory
            Get
                Throw New InvalidOperationException
            End Get
            Set(ByVal Value As Directory)
                Throw New InvalidOperationException
            End Set
        End Property

        Public ReadOnly Property ControlChannel() As ControlChannel Implements ISessionState.ControlChannel
            Get
                Throw New InvalidOperationException
            End Get
        End Property

        Public ReadOnly Property IsBusy() As Boolean Implements ISessionState.IsBusy
            Get
                Throw New InvalidOperationException
            End Get
        End Property

        Public ReadOnly Property RootDirectory() As Directory Implements ISessionState.RootDirectory
            Get
                Throw New InvalidOperationException
            End Get
        End Property

        Public Sub AbortTransfer() Implements ISessionState.AbortTransfer

            Throw New InvalidOperationException

        End Sub

        Public Sub Close() Implements ISessionState.Close

            ' Nothing to do - don't want to throw an error...

        End Sub

    End Class

End Namespace

File: Source/OldCode/Archive/2005 TVA Code Library/Version/CS/TVA.Core_CS/Net/Ftp/SessionDisconnected.cs
using System.Diagnostics;
using System.Linq;
using System.Data;
using System.Collections;
using Microsoft.VisualBasic;
using System.Collections.Generic;
using System;
using System.Text;

// James Ritchie Carroll - 2003


namespace TVA
{
	namespace Net
	{
		namespace Ftp
		{
			
			
			internal class SessionDisconnected : ISessionState
			{
				
				
				
				private Session m_host;
				private string m_server;
				private int m_port;
				private bool m_caseInsensitive;
				
				internal SessionDisconnected(Session h, bool CaseInsensitive)
				{
					
					m_port = 21;
					m_host = h;
					m_caseInsensitive = CaseInsensitive;
					
				}
				
				public string Server
				{
					get
					{
						return m_server;
					}
					set
					{
						m_server = value;
					}
				}
				
				public int Port
				{
					get
					{
						return m_port;
					}
					set
					{
						m_port = value;
					}
				}
				
				public void Connect(string UserName, string Password)
				{
					
					ControlChannel ctrl = new ControlChannel(m_host);
					
					ctrl.Server = m_server;
					ctrl.Port = m_port;
					ctrl.Connect();
					
					try
					{
						ctrl.Command("USER " + UserName);
						
						if (ctrl.LastResponse.Code == Response.UserAcceptedWaitingPass)
						{
							ctrl.Command("PASS " + Password);
						}
						
						if (ctrl.LastResponse.Code != Response.UserLoggedIn)
						{
							throw (new AuthenticationException("Failed to login.", ctrl.LastResponse));
						}
						
						m_host.State = new SessionConnected(m_host, ctrl, m_caseInsensitive);
						((SessionConnected) m_host.State).InitRootDirectory();
					}
					catch
					{
						ctrl.Close();
						throw;
					}
					
				}
				
				public Directory CurrentDirectory
				{
					get
					{
						throw (new InvalidOperationException);
					}
					set
					{
						throw (new InvalidOperationException);
					}
				}
				
				public ControlChannel ControlChannel
				{
					get
					{
						throw (new InvalidOperationException);
					}
				}
				
				public bool IsBusy
				{
					get
					{
						throw (new InvalidOperationException);
					}
				}
				
				public Directory RootDirectory
				{
					get
					{
						throw (new InvalidOperationException);
					}
				}
				
				public void AbortTransfer()
				{
					
					throw (new InvalidOperationException);
					
				}
				
				public void Close()
				{
					
					// Nothing to do - don't want to throw an error...
					
				}
				
			}
			
		}
	}
}
